src/main/java/graphql/annotations/processor/retrievers/GraphQLAdditionalTypesHandler.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.processor.retrievers;
+
+import graphql.GraphQLException;
+import graphql.annotations.annotationTypes.GraphQLTypeResolver;
+import graphql.annotations.processor.ProcessingElementsContainer;
+import graphql.annotations.processor.searchAlgorithms.SearchAlgorithm;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLType;
+import org.clapper.util.classutil.ClassFilter;
+import org.clapper.util.classutil.ClassFinder;
+import org.clapper.util.classutil.ClassInfo;
+import org.clapper.util.classutil.SubclassClassFilter;
+import org.osgi.service.component.annotations.Component;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static graphql.annotations.processor.util.ObjectUtil.getAllFields;
+
+@SuppressWarnings("ConstantConditions")
+@Component(service = GraphQLAdditionalTypesHandler.class, immediate = true)
+public class GraphQLAdditionalTypesHandler {
+
+    private GraphQLInterfaceRetriever graphQLInterfaceRetriever;
+    private GraphQLObjectInfoRetriever graphQLObjectInfoRetriever;
+    private SearchAlgorithm methodSearchAlgorithm;
+    private SearchAlgorithm fieldSearchAlgorithm;
+    private GraphQLObjectInfoRetriever infoRetriever;
+    private ClassFinder classFinder;
+
+    public GraphQLAdditionalTypesHandler() {
+        classFinder = new ClassFinder();
+        classFinder.addClassPath();
+        infoRetriever = new GraphQLObjectInfoRetriever();
+    }
+
+    public Set<GraphQLType> getAdditionalInterfacesImplementations(Class<?> root, ProcessingElementsContainer container) {
+        Set<GraphQLType> additionalFields = new HashSet<>();
+
+        additionalFields.addAll(getNewImplementations(container, classFinder, root));
+        for (Method method : graphQLObjectInfoRetriever.getOrderedMethods(root)) {
```

Comment:
In my opinion we should not go through the different fields but rather let the user explicitely pass the interfaces he wants to parse. I don't really see the benefit and it could be confusing and error-prone, as you don't necessarily want to consider all fields as interfaces with automatically discovered implementations.

Reply:
The problem is the following:
We build the schema like a tree, starting from query and go to its field and so on. If there is an implementation of some interface that is not in the schema (only the interface is in the schema), I will not get to that implementation class, so even with your solution the problem will still exists. The only way I can think of finding the classs is to search the class path for all the classes that implement the interface. For more details about the problem, take a look [here](https://github.com/graphql-java/graphql-java/issues/950)

Reply:
I don't understand what you mean. Do you want to have an attribute in `@GraphQLTypeResolver` that says to search for implementation, So if it is off we don't do it? 
Well, I think I can add this attribute, but I also think that in most cases people will set this attribute to "true", because if you have an interface, its implementations will not be explicitly in the schema. But I don't mind adding that

Reply:
Until now I was adding the implementation manually - so at least for compatibility, or if we explicitely do not want to parse all classes, the attribute could be useful. Default value can be to do the parsing, no problem with that, in most cases it should really help.
What about calling it automatically everytime we create a new type ?